src/transaction/locator_sr.c

```diff
@@ -6274,12 +6285,35 @@ locator_delete_force_internal (THREAD_ENTRY * thread_p, HFID * hfid, OID * oid,
       /* remove query result cache entries which are relevant with this class */
       if (!QFILE_IS_LIST_CACHE_DISABLED)
 	{
+	  OID superclass_oid;
+	  OID real_class_oid;
+	  HFID real_hfid;
+
 	  if (qexec_clear_list_cache_by_class (thread_p, &class_oid) != NO_ERROR)
 	    {
 	      er_log_debug (ARG_FILE_LINE,
 			    "locator_delete_force: qexec_clear_list_cache_by_class failed for class { %d %d %d }\n",
 			    class_oid.pageid, class_oid.slotid, class_oid.volid);
 	    }
+
+	  OID_SET_NULL (&superclass_oid);
+
+	  HFID_COPY (&real_hfid, hfid);
+	  COPY_OID (&real_class_oid, &class_oid);
+
+	  error_code =
+	    partition_prune_update (thread_p, &class_oid, &copy_recdes, NULL, DB_PARTITIONED_CLASS, &real_class_oid,
```

Comment:
Can't the routine be executed only in the case of a partition table? like below.
```
locator_update_force()
if (pruning_type != DB_NOT_PARTITIONED_CLASS)
```

Reply:
the pruning_type is necessary but the caller doesn't transfer the value. the partition_load_pruning_context called by partition_prun_update  returns null as superclass_oid when the pruning_type is not DB_PARTITION_CLASS. To modify the function call locator_delete_force with pruning_type as an argument needs to modify several routines.
In addition, partition_prune_update is only called at QUERY_CACHE is enabled. Therefore the partition_prune_update 
 calling cost is limited.